src/sharepoint/rest/actionables.ts

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+import { Queryable } from "./queryable";
+import * as Mixins from "./mixins";
+import * as Util from "../../utils/util";
+
+/**
+ * A Queryable which only exposes the get method
+ * 
+ */
+export class Gettable<T> extends Queryable implements Mixins.Gettable {
+
+    /**
+     * Creates a new instance of the Gettable class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this actionable
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl);
+    }
+
+    /**
+     * Execute the get request
+     * 
+     */
+    public get(): Promise<T> { return; }
+}
+Util.applyMixins(Gettable, Mixins.Gettable);
+
+
+/**
+ * A Queryable which only exposes the get and select methods
+ * 
+ */
+export class SelectableGettable<T> extends Queryable implements Mixins.Gettable, Mixins.Selectable {
+
+    /**
+     * Creates a new instance of the SelectableGettable class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this actionable
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl);
+    }
+
+    /**
+     * Select the fields to return
+     * 
+     * @param selects One or more fields to return
+     */
+    public select(...selects: string[]): Gettable<T> { return; }
+
+    /**
+     * Execute the get request
+     * 
+     */
+    public get(): Promise<T> { return; }
+}
+Util.applyMixins(SelectableGettable, Mixins.Gettable, Mixins.Selectable);
+
+
+/**
+ * A Queryable which only exposes the get and filter methods
+ * 
+ */
+export class FilterableGettable<T> extends Queryable implements Mixins.Gettable, Mixins.Filterable {
+
+    /**
+     * Creates a new instance of the FilterableGettable class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this actionable
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl);
+    }
+
+    /**
+     * Applies a filter to the request
+     * 
+     * @param filter The filter string (docs: https://msdn.microsoft.com/en-us/library/office/fp142385.aspx)
+     */
+    public filter(filter: string): Gettable<T> { return; }
+
+    /**
+     * Execute the get request
+     * 
+     */
+    public get(): Promise<T> { return; }
+}
+Util.applyMixins(FilterableGettable, Mixins.Gettable, Mixins.Filterable);
+
+
+/**
+ * A Queryable which only exposes the get, select and filter methods
+ * 
+ */
+export class FilterableSelectableGettable<T> extends Queryable implements Mixins.Gettable, Mixins.Selectable, Mixins.Filterable {
```

Comment:
These types are there to provide a terminus for the request building. In some cases you end up in a place where there is no more building to do and our objects should just allow "get" for example. So these are there to support the intellisense. I do not think we will or should end up with every combination since that would be a mess. I also looked while I was doing this at doing away with the mixins entirely and going to sub classes that were a combination of the various mixin methods, but decided that would end up being a mess as well. So far I like the mixin pattern of adding behaviors to the appropriate classes.